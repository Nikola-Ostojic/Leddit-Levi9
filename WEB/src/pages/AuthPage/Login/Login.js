import React from "react";
import { withRouter } from "react-router-dom";
import { Button, TextField } from "@material-ui/core";

export class Login extends React.Component {
  state = {
    password: "",
    email: ""
  };

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  submit = (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    const { login } = this.props;

    login(email, password);
  };

  render() {
    const { email, password } = this.state;
    const { error } = this.props;
    return (
      <div>
        <h3>Login</h3>
        <p>To make our lives easier</p>
        <p>user@levi9.com -- User</p>
        <p>admin@levi9.com -- Admin</p>
        <form onSubmit={this.submit}>
          <TextField
            fullWidth
            margin="normal"
            name="email"
            type="email"
            required
            label="Email"
            onChange={this.handleChange}
            value={email}
          />
          <TextField
            fullWidth
            margin="normal"
            name="password"
            type="password"
            required
            onChange={this.handleChange}
            label="Password"
            value={password}
          />
          <br />
          <br />
          <Button color="primary" variant="contained" type="submit">
            Login
          </Button>
          {error && <p>{error}</p>}
        </form>
      </div>
    );
  }
}

export default withRouter(Login);
