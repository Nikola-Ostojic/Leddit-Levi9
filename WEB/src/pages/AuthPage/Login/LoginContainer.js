import { connect } from "react-redux";

import { getError } from "modules/Auth/AuthSelectors";
import { login } from "modules/Auth/AuthActions";

import Login from "./Login";

export const mapStateToProps = (state) => ({
  error: getError(state)
});

export default connect(mapStateToProps, { login })(Login);
