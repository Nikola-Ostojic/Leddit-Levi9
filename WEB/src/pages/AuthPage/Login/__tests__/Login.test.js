import React from "react";
import { shallow } from "enzyme";
import { TextField } from "@material-ui/core";

import { Login } from "../Login";

describe("Login component tests", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should match snapshot with error", () => {
    expect(shallow(<Login error="An error happened" />)).toMatchSnapshot();
  });

  it("Should call login if the required fields are filled out in the form", () => {
    // Arrange
    const login = jest.fn();
    const wrapper = shallow(<Login login={login} />);
    const email = "Test@gmail.com";
    const password = "Test";

    // Act
    wrapper.setState({ email, password });

    const form = wrapper.find("form");
    form.simulate("submit", {
      preventDefault: jest.fn()
    });

    // Assert
    expect(login).toBeCalledWith(email, password);
  });

  it("Should onChange happen handleChange should appropriately update the state", () => {
    // Arrange
    const wrapper = shallow(<Login />);
    const changeValue = "user@levi9.com";
    const changeEvent = { target: { value: changeValue, name: "email" } };
    const setStateSpy = jest.spyOn(wrapper.instance(), "setState");

    // Act
    const emailInputField = wrapper.find(TextField).at(0);
    emailInputField.simulate("change", changeEvent);

    // Assert
    expect(setStateSpy).toBeCalledWith({ email: changeValue });
    expect(wrapper.state()).toEqual({
      password: "",
      email: changeValue
    });
  });
});
