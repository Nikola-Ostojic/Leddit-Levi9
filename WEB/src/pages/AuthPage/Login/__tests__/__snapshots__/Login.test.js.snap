// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests Should match snapshot 1`] = `
<div>
  <h3>
    Login
  </h3>
  <p>
    To make our lives easier
  </p>
  <p>
    user@levi9.com -- User
  </p>
  <p>
    admin@levi9.com -- Admin
  </p>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Email"
      margin="normal"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Password"
      margin="normal"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <br />
    <br />
    <WithStyles(ForwardRef(Button))
      color="primary"
      type="submit"
      variant="contained"
    >
      Login
    </WithStyles(ForwardRef(Button))>
  </form>
</div>
`;

exports[`Login component tests Should match snapshot with error 1`] = `
<div>
  <h3>
    Login
  </h3>
  <p>
    To make our lives easier
  </p>
  <p>
    user@levi9.com -- User
  </p>
  <p>
    admin@levi9.com -- Admin
  </p>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Email"
      margin="normal"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      label="Password"
      margin="normal"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <br />
    <br />
    <WithStyles(ForwardRef(Button))
      color="primary"
      type="submit"
      variant="contained"
    >
      Login
    </WithStyles(ForwardRef(Button))>
    <p>
      An error happened
    </p>
  </form>
</div>
`;
